### ISSUE TITLE
Allow complete content of dialog to be passed to the service

### ISSUE THREAD

**Issue by gyoshev** [telerik/kendo-angular#1046]
## I'm submitting a...
* Suggestion for improvement

## Current behavior
The current dialog service API works great for simple text dialogs. However, passing custom actions is involved, and follows a strange pattern. There is no way to keep the dialog content and actions together when opening a dialog from a service.

## Expected behavior
Ideally, the dialog service should be able to open a dialog from a defined component:

```ts
class AppComponent {
  constructor( private dialogService: DialogService ) {}

  public showCustomDialog() {
      const dialog = this.dialogService.open(CustomDialogComponent);
      dialog.result.subscribe(res => console.log(res));
  }
}
```

```ts
@Component({
    template: `
        <kendo-dialog-title>Custom title</kendo-dialog-title>

        Custom dialog content.

        <kendo-dialog-actions>
              <button kendoButton (click)="dialog.close({ result: false })">No</button>
              <button kendoButton (click)="dialog.close({ result: true })" [primary]="true">Yes</button>
        </kendo-dialog-actions>
    `
})
class CustomDialogComponent {
    constructor( public dialog: DialogRef ) {}
}
```

## What is the motivation or use case for changing the behavior?
Improve the usability of the dialog service for custom dialogs.

**Comment by meriturva** [comment]
I think it is really important to have same approach using component or template (same css customisation here for actions). Right now i'm implementing dialog component with dialogService and i just would like to use common approach.

**Comment by HolgerPersch** [comment]
Is there an estimation when this feature will be available?

**Comment by rjpackito** [comment]
We need this kind of approach.

**Comment by HolgerPersch** [comment]
Shall we file a feature request at the [feedback portal](http://kendoui-feedback.telerik.com/forums/555517-kendo-ui-for-angular-2-feedback)?

**Comment by dtopalov** [comment]
Hi guys,

You can check out the [development version](https://www.telerik.com/kendo-angular-ui-develop/components/installation/development-builds/) of the package that contains the discussed enhancement:

[https://www.telerik.com/kendo-angular-ui-develop/components/dialogs/dialog/service/#toc-passing-title-content-and-actions-as-a-single-component](https://www.telerik.com/kendo-angular-ui-develop/components/dialogs/dialog/service/#toc-passing-title-content-and-actions-as-a-single-component)

**Comment by dtopalov** [comment]
Released in [3.10.0](https://www.telerik.com/kendo-angular-ui-develop/components/dialogs/changelog/)
